File: web/schemas/admin/payment_providers.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class PaymentProviderResponse(BaseModel):
    id: int
    provider_key: str
    display_name: str
    is_enabled: bool
    sort_order: int
    updated_at: Optional[datetime]

    model_config = {"from_attributes": True}


class PaymentProviderUpdateRequest(BaseModel):
    is_enabled: Optional[bool] = None
    sort_order: Optional[int] = None
    display_name: Optional[str] = None


class PaymentProvidersListResponse(BaseModel):
    items: list[PaymentProviderResponse]

File: frontend/src/api/admin/payment-providers.ts
import { apiRequest } from '@/api/client'

export interface PaymentProviderResponse {
  id: number
  provider_key: string
  display_name: string
  is_enabled: boolean
  sort_order: number
  updated_at: string | null
}

export interface PaymentProvidersListResponse {
  items: PaymentProviderResponse[]
}

export interface PaymentProviderUpdateRequest {
  is_enabled?: boolean
  sort_order?: number
  display_name?: string
}

export function getAdminPaymentProviders(): Promise<PaymentProvidersListResponse> {
  return apiRequest<PaymentProvidersListResponse>('/admin/payment-providers')
}

export function updatePaymentProvider(
  providerId: number,
  body: PaymentProviderUpdateRequest,
): Promise<PaymentProviderResponse> {
  return apiRequest<PaymentProviderResponse>(`/admin/payment-providers/${providerId}`, {
    method: 'PATCH',
    body: JSON.stringify(body),
  })
}
